### PR TITLE
Manual dependency Bump

### PR DESCRIPTION
**Pull request by sharathsivakumar**
Dependabot has raised the following updates:

- [ ] https://github.com/etcd-io/etcd/pull/16103 . This is a dependency for [prometheus client_golang](https://github.com/prometheus/client_golang/blob/main/go.mod). Upgraded to v0.10.0 as needed by prometheus client_golang
- [x] https://github.com/etcd-io/etcd/pull/16104. 
- [x] https://github.com/etcd-io/etcd/pull/16107
- [x] https://github.com/etcd-io/etcd/pull/16108
- [x] https://github.com/etcd-io/etcd/pull/16109

### CODE REVIEW

**Comment by sharathsivakumar** [comment]
@ahrtr @jmhbnz This is ready for review. Please take a look and let me know if I need to fix anything.

File: tools/mod/go.mod [hunk 1]
```diff
@@ -79,7 +79,7 @@ require (
 	golang.org/x/exp/typeparams v0.0.0-20221208152030-732eee02a75a // indirect
 	golang.org/x/mod v0.10.0 // indirect
 	golang.org/x/net v0.11.0 // indirect
-	golang.org/x/sync v0.2.0 // indirect
+	golang.org/x/sync v0.3.0 // indirect
```
Comment (ahrtr) [inline tools/mod/go.mod:82]: The [test](https://github.com/etcd-io/etcd/blob/120f1d6d058689d42f9f6c7b0aafddcb623f436f/tests/go.mod#L43) directly depends on `golang.org/x/sync`, please also bump it for the test.

Comment (sharathsivakumar) [reply tools/mod/go.mod:82]: Apologies. I missed the tests. This is done now. @ahrtr .

File: tools/mod/go.mod [hunk 2]
```diff
@@ -77,7 +77,7 @@ require (
 	github.com/zmap/zlint/v3 v3.4.1 // indirect
 	golang.org/x/crypto v0.10.0 // indirect
 	golang.org/x/exp/typeparams v0.0.0-20221208152030-732eee02a75a // indirect
-	golang.org/x/mod v0.10.0 // indirect
+	golang.org/x/mod v0.11.0 // indirect
```
Comment (ahrtr) [inline tools/mod/go.mod:80]: It's a pure indirect dependency. Usually we don't bump pure indirect dependency. Please read [#indirect-dependencies](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/dependency_management.md#indirect-dependencies)

File: tools/mod/go.mod [hunk 3]
```diff
@@ -83,7 +83,7 @@ require (
 	golang.org/x/sys v0.9.0 // indirect
 	golang.org/x/term v0.9.0 // indirect
 	golang.org/x/text v0.10.0 // indirect
-	golang.org/x/tools v0.9.1 // indirect
+	golang.org/x/tools v0.10.0 // indirect
```
Comment (ahrtr) [inline tools/mod/go.mod:86]: same as above. Please read [#indirect-dependencies](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/dependency_management.md#indirect-dependencies)

File: tools/mod/go.mod (hunk 2 above)
Comment (sharathsivakumar) [reply tools/mod/go.mod:80]: Ah understood. I bumped this as  tools module does not depend on any other modules.

File: tools/mod/go.mod (hunk 3 above)
Comment (sharathsivakumar) [reply tools/mod/go.mod]: done!

File: tools/mod/go.mod (hunk 2 above)
Comment (sharathsivakumar) [reply tools/mod/go.mod]: Done!

**Comment by ahrtr** [comment]
could you squash the commits to 2 (preferred) or 1 commits?

**Comment by sharathsivakumar** [comment]
You beat me to it. I was about to ask the same. I will squash into 2 commits, 1 for each module.

**Comment by ahrtr** [comment]
> 1 for each module.

one commit for each dependency. In this example, one commit for `github.com/prometheus/client_golang`, and the other commit for `golang.org/x/sync`

**Comment by ahrtr** [comment]
If you want to keep it simple, squashing all into just one commit is also accepted in this PR.

**Comment by sharathsivakumar** [comment]
@ahrtr Its done. I have added 2 commits for each of the dependencies.

File: tests/go.mod [hunk 4]
```diff
@@ -40,7 +40,7 @@ require (
 	go.opentelemetry.io/proto/otlp v0.19.0
 	go.uber.org/zap v1.24.0
 	golang.org/x/crypto v0.10.0
-	golang.org/x/sync v0.2.0
+	golang.org/x/sync v0.3.0
```
Comment (ahrtr) [inline tests/go.mod:43]: You should get this included in the first commit. But I won't want to insist on it in this PR.

Comment (jmhbnz) [reply tests/go.mod:43]: Yeah my preference would be to address this before merge if possible.

**Review by jmhbnz — APPROVED** [review]
Thanks for your help handling dependencies this week @sharathsivakumar.

Overall LGTM 👍🏻

File: tests/go.mod (hunk 4 above)
Comment (ahrtr) [reply tests/go.mod:43]: It isn't a big deal, let's merge it for now.

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @sharathsivakumar

File: tests/go.mod (hunk 4 above)
Comment (sharathsivakumar) [reply tests/go.mod:43]: Noted. I will take care of this going forward.

**Pull request merged**